src/widgets/barchart/mod.rs

```diff
@@ -122,12 +120,39 @@ impl<'a> BarChart<'a> {
         self
     }
 
+    pub fn group_gap(mut self, gap: u16) -> BarChart<'a> {
+        self.group_gap = gap;
+        self
+    }
+
     pub fn style(mut self, style: Style) -> BarChart<'a> {
         self.style = style;
         self
     }
 }
 
+impl<'a> BarChart<'a> {
+    fn remove_unvisible_groups_and_bars(&mut self, mut width: u16) {
+        for group_index in 0..self.data.len() {
+            let n_bars = self.data[group_index].bars.len() as u16;
+            let bars_width = n_bars * self.bar_width + n_bars.saturating_sub(1) * self.bar_gap;
+
+            if width > bars_width {
+                width = width.saturating_sub(bars_width + self.group_gap);
```

Comment:
You seem to be underestimating the width required to render the bar chart by one for each group. This results in the bar chart rendering over the border when resizing the terminal.

The following suggestion would fix it, but I haven't looked too much at `render_widget` so this might be something you want to change there.

```suggestion
                width = width.saturating_sub(bars_width + self.group_gap + 1);
```